src/client/datascience/notebook/outputs/rendererCommunication.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import { notebooks } from 'vscode';
+import { IExtensionSyncActivationService } from '../../../activation/types';
+import { UseVSCodeNotebookEditorApi } from '../../../common/constants';
+import { disposeAllDisposables } from '../../../common/helpers';
+import { IDisposable } from '../../../common/types';
+import { noop } from '../../../common/utils/misc';
+import { PlotSaveHandler } from './plotSaveHandler';
+
+const rendererId = 'jupyter-notebook-renderer';
+type RendererMessageTypes = { type: 'saveAs'; outputId: string; mimnType: string };
+
+@injectable()
+export class RendererCommunication implements IExtensionSyncActivationService, IDisposable {
+    private readonly disposables: IDisposable[] = [];
+    constructor(
+        @inject(UseVSCodeNotebookEditorApi) private readonly useVSCNotebook: boolean,
+        @inject(PlotSaveHandler) private readonly plotSaveHandler: PlotSaveHandler
+    ) {}
+
+    public dispose() {
+        disposeAllDisposables(this.disposables);
+    }
+    public activate() {
+        if (!this.useVSCNotebook) {
+            return;
+        }
+
+        const api = notebooks.createRendererMessaging<unknown, RendererMessageTypes>(rendererId);
+        api.onDidReceiveMessage(
+            ({ editor, message }) => {
+                if (message.type === 'saveAs') {
+                    this.plotSaveHandler.savePlot(editor, message.outputId, message.mimnType).catch(noop);
```

Comment:
```suggestion
                    this.plotSaveHandler.savePlot(editor, message.outputId, message.mimeType).catch(noop);
```

Reply:
Hm, maybe I'm missing something. Where does the corresponding `postMessage` happen? I'm not seeing it in this PR. Though I assume it comes from the renderers, so maybe that change happens in https://github.com/microsoft/vscode-notebook-renderers?

Reply:
Never mind, just read your issue description 🤡 

> Corresponding change microsoft/vscode-notebook-renderers#55